src/vs/workbench/contrib/positronHistory/browser/components/historyEntry.tsx

```diff
@@ -0,0 +1,621 @@
+			onHeightChange(height);
+		}
+	}, [isSelected, colorizedHtml, smartExcerpt, onHeightChange]);
+
+	/**
+	 * Apply font info after render
+	 */
+	useEffect(() => {
+		if (codeRef.current) {
+			applyFontInfo(codeRef.current, props.fontInfo);
+		}
+	});
+
+	/**
+	 * Handle context menu
+	 */
+	const handleContextMenu = (event: React.MouseEvent) => {
+		event.preventDefault();
+		event.stopPropagation();
+
+		const x = event.clientX;
+		const y = event.clientY;
+
+		const actions: IAction[] = [];
+
+		// Add "Send to Console" action
+		actions.push({
+			id: 'positronHistory.sendToConsole',
+			label: positronHistorySendToConsole,
+			tooltip: '',
+			class: undefined,
+			enabled: true,
+			run: () => onToConsole()
+		});
+
+		// Add "Send to Editor" action
+		actions.push({
+			id: 'positronHistory.sendToEditor',
+			label: positronHistorySendToEditor,
+			tooltip: '',
+			class: undefined,
+			enabled: true,
+			run: () => onToSource()
+		});
+
+		// Add separator
+		actions.push(new Separator());
+
+		// Add "Copy" action
+		actions.push({
+			id: 'positronHistory.copy',
+			label: positronHistoryCopy,
+			tooltip: '',
+			class: undefined,
+			enabled: true,
+			run: () => onCopy()
+		});
+
+		// Add separator
+		actions.push(new Separator());
+
+		// Add "Delete" action
+		actions.push({
+			id: 'positronHistory.delete',
+			label: positronHistoryDeleteEntry,
+			tooltip: '',
+			class: undefined,
+			enabled: true,
+			run: () => onDelete()
+		});
+
+		// Show the context menu
+		services.contextMenuService.showContextMenu({
+			getActions: () => actions,
+			getAnchor: () => ({ x, y }),
+			anchorAlignment: AnchorAlignment.LEFT,
+			anchorAxisAlignment: AnchorAxisAlignment.VERTICAL
+		});
+	};
+
+	const showExpandButton = lineCount > MAX_COLLAPSED_LINES;
+	const needsTruncation = showExpandButton && !isSelected && !smartExcerpt;
+	const codeToDisplay = isSelected ? entry.input : truncateCode(entry.input, MAX_COLLAPSED_LINES);
+
+	// Determine truncation state for CSS classes
+	const hasTruncatedTop = smartExcerpt && smartExcerpt.hiddenAbove > 0;
+	const hasTruncatedBottom = needsTruncation || (smartExcerpt && smartExcerpt.hiddenBelow > 0);
+
+	// Override the height from react-window's style to allow natural content sizing
+	const styleWithoutHeight = { ...style, height: 'auto' };
+
+	return (
+		<div
+			ref={entryRef}
+			className={`history-entry ${isSelected ? (hasFocus ? 'selected' : 'selected-unfocused') : ''}`}
+			style={styleWithoutHeight}
+			onContextMenu={handleContextMenu}
+			onDoubleClick={() => onToConsole()}
+			onMouseDown={(e) => {
+				// Use onMouseDown instead of onClick to ensure selection happens before focus events
+				// This prevents the two-click issue when the panel is unfocused
+				if (e.button === 0) { // Only handle left clicks
+					e.preventDefault(); // Prevent default focus behavior that steals focus
+					onSelect();
+				}
+			}}
+		>
+			<div className="history-entry-content">
```

Comment:
super minor, but using double quotes for classnames normally throws an eslint error. These should be single quotes.
```suggestion
			<div className='history-entry-content'>
```

Reply:
this should now be addressed!